app/controllers/api/user_blocks_controller.rb

```diff
@@ -14,5 +16,30 @@ def show
     rescue ActiveRecord::RecordNotFound
       raise OSM::APINotFoundError
     end
+
+    def create
+      raise OSM::APIBadUserInput, "No user was given" unless params[:user]
+
+      user = User.visible.find_by(:id => params[:user])
+      raise OSM::APINotFoundError unless user
+      raise OSM::APIBadUserInput, "No reason was given" unless params[:reason]
+      raise OSM::APIBadUserInput, "No period was given" unless params[:period]
+
+      period = Integer(params[:period], :exception => false)
+      raise OSM::APIBadUserInput, "Period one is in a wrong format" unless period
+
+      max_period = UserBlock::PERIODS.max
+      raise OSM::APIBadUserInput, "Period must be between 0 and #{max_period}" if period.negative? || period > max_period
+      raise OSM::APIBadUserInput, "Needs_view must be true if provided" unless params[:needs_view].nil? || params[:needs_view] == "true"
+
+      @user_block = UserBlock.create(
```

Comment:
We should probably check if the user to be blocked is not a mod or admin and is also different from the current user id.

Reply:
> What should happen if the user to be blocked is already currently blocked? Or if we're trying to POST the same user block twice, and introduce duplicates.

You can create multiple blocks from the web interface.

Currently we have two qualitatively different block types: one that requires reading the message and one that doesn't. In the future there might be more types. It might be useful to have multiple active blocks with different periods at the same time.

Reply:
> We should probably check if the user to be blocked is not a mod or admin (and implicitly is also different from the current user id).

In case someone who is not supposed to have moderator permissions gets a token with `write_blocks`, blocking mods/admins is the last thing they'll want to do. This will be noticed very quickly. They'll be basically announcing themselves "I stole a mod token!"

Reply:
They might notice it quickly. However, an attacker might create hundreds or thousands of user blocks for mod accounts, so it would require an admin to revoke all user blocks and make the mod accounts usable again.

Reply:
we can add a "revoke all blocks from this account" button to the web ui

Reply:
Yes, that would also work, and would be a topic for another PR.

Reply:
"revoke all blocks" is added